File: java/org/gessinger/gepard/xmp/Responder.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;

import java.util.HashMap ;

public class Responder
{
  static public void main ( String[] args )
  {
    try
    {
      Responder j = new Responder() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  Responder()
  throws Exception
  {
    final Client client = Client.getInstance() ;
    client.onShutdown ( new InfoCallback()
    {
      public void info ( Client c, Event e )
      {
        System.out.println ( e ) ;
        System.exit ( 0 ) ;
      }
    });
    String name = "getFileList" ;
    System.out.println ( "Listen for requests with name=" + name ) ;
    client.on ( name, new EventListener()
    {
      public void event ( Event e )
      {
        String[] fileList = new String[] { "a.java", "b.java", "c.java" } ;
        System.out.println ( "Request in" ) ;
        System.out.println ( "File list out:\n" + Util.toString ( fileList ) ) ;
        e.putBodyValue ( "file_list", fileList ) ;
        try
        {
          e.sendBack() ;
        }
        catch ( Exception exc )
        {
          System.out.println ( Util.toString ( exc ) ) ;
        }
      }
    } ) ;
  }
}

File: python/xmp/Responder.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )

import gepard

client = gepard.Client.getInstance()
# client.setDaemon ( True )
def on_close ( err, info ):
	print ( err )
def on_error ( err, info ):
	print ( err )
def on_shutdown ( err, info ):
	print ( "shutdown called" )

client.onClose ( on_close )
client.onError ( on_error )
client.onShutdown ( on_shutdown )

print ( "Listen for requests with name=getFileList" ) ;

fileList = [ "a.py", "b.py", "c.py" ] ;
def on_getFileList ( event ):
	print ( "Request in" ) ;
	print ( "File list out:" ) ;
	print ( fileList ) ;
	event.body["file_list"] = fileList ;
	event.sendBack() ;

client.on ( "getFileList", on_getFileList )

gepard.util.exitWithSIGINT()
